src/org/rocketproplab/marginalstability/flightcomputer/hal/LSM9DS1Mag.java

```diff
@@ -0,0 +1,392 @@
+package org.rocketproplab.marginalstability.flightcomputer.hal;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayDeque;
+
+import org.rocketproplab.marginalstability.flightcomputer.ErrorReporter;
+import org.rocketproplab.marginalstability.flightcomputer.Errors;
+import org.rocketproplab.marginalstability.flightcomputer.math.Vector3;
+
+import com.pi4j.io.i2c.I2CDevice;
+
+/**
+ * The HAL implementation for the LSM9DS1 magnetometer sensor. Datasheet can be
+ * found here: <a href="https://www.st.com/resource/en/datasheet/lsm9ds1.pdf">https://www.st.com/resource/en/datasheet/lsm9ds1.pdf</a><br>
+ * 
+ * Every time poll is called, this sensor will acquire 1 sample from the LSM9DS1
+ * magnetometer registers. Each sample gets put into a queue which can be accessed
+ * with the {@link #getNext()} method. Whether or not the queue is empty can be
+ * read using the {@link #hasNext()} method. It is recommended to call
+ * {@link #hasNext()} before each call of {@link #getNext()}.
+ * 
+ * @author Enlil Odisho
+ *
+ */
+public class LSM9DS1Mag implements PollingSensor, IMU {
```

Comment:
The IMU interface has lost its meaning since we split Mag & Accel/Gyro. I think separate interfaces would be a better solution to provide the HAL.